### DISCUSSION TITLE
Lolin ESP32 S2 issue?

### DISCUSSION THREAD

**Discussion by martin072** [arendst/Tasmota#19900] [Q&A] [answered]
Hi,

I've noticed something really strange. I flashed a Lolin ESP32 S2 with tasmota 13.1.0 and noticed that some i2c sensors are not detected correctly. For one a SCD40 was detect during boot, but no sensor information was showing. A AHT21+BMP sensor was initially detected as a Luminance sensor instead  of Temp/Hum/Press and showed sensor info regarding light values. As second ATH21 (as I thought the first one could be broken) showed the BMP sensor correct, but the AHT21 was detected as Luminance as well. 
It's an own compiled binary, however the same  compiled for an ESP32 Dev board works perfectly fine with all the above named sensors. Could this be anything related to the S2 ? 

(At the moment I am not near the project, this weekend I can supply some screen shots if so required)

**Comment by sfromis** [comment]
A general recommendation is to [disable I2C drivers](https://tasmota.github.io/docs/I2CDEVICES/) for devices included in the binary, but not hooked up. Not doing that sometimes lead to the "wrong driver" grabbing a device, even though you also have the right driver included. Getting a luminance sensor detected sounds a *lot* like this, and at boot time you can see what sensors were detected, correctly or not.

A risk factor is that you got a clone of "lesser quality" than the actually original branded Lolin product, I see "wemos.cc" on the back of mine. SCD40 can use up to 205 mA, which may overwhelm penny-pinched devices.

**Reply by martin072** [reply]
Ok, thanks for that tip, for many years i've used a 'general' compiled binary for all my projects and seemed to work fine. perhaps something added some sensors in 13.x that changed it? Nevertheless I will go thru them and disable what I do not use. 

Yes they should be Wemos, as  they were bought thru the official store.

About the SCD40, as long as the power supply is sufficient, that should not be an issues correct? or does it has effects on the i2c bus?

**Comment by martin072** [comment]
That seemed to get the sensors displayed as it should, however I've noticed this:

```
BMP280 Temperature | -141.8 °C
BMP280 Pressure | 1485.3 hPa
AHT2X Temperature | 18.5 °C
AHT2X Humidity | 59.2 %
AHT2X Dew point | 10.4 °C
```

The BMP sensor seems to show incorrect values every now and then for a while, then returning to normal values and then incorrect etc etc. Is this a faulty module, or proof of cheap AliEpress quality? or something else?

```weblog 4``` does not show any errors or other issues

**Reply by sfromis** [reply]
To me, that might be suggestive of of power issues. While I can't say anything for sure, my impression is that the Bosch sensors are more likely do do that if not getting good stable power. Not sure if the power demands of the SCD40 could have spikes starving the Bosch sensor, but my impression is again that other sensors drawing power could be a factor.

**Comment by martin072** [comment]
So the SCD40 is not connected, just the AHTX + BMP (2 in one). Will look into the power issue. At the moment it's powered by a 2A supply so that would be very odd..

**Reply by sfromis** [reply]
What I meant has nothing to do with the amperage of the supply (unless very low or uneven), but dynamic effects of drawing uneven power. And phone "chargers" may be bad at delivering smooth power, that is not important for their primary use case.

**Reply by martin072** [reply]
Ah ok.

**Reply by martin072** [reply]
So, had a look with a (not professional) scope, and shows some voltage drops across the 5v and 3.3v (although not consistent). 
the 12v seemed stable with no drops, so that leads me to the 12->5v circuit perhaps? any idea what could cause this (see pics)?

![7](https://github.com/arendst/Tasmota/assets/24524506/2c895ea5-c634-437e-8be0-03c754f05106)

Schematic
<img width="847" alt="Screenshot 2023-11-05 at 10 44 52" src="https://github.com/arendst/Tasmota/assets/24524506/b8013517-20fd-4fac-af06-3166d6c682a9">

**Reply by sfromis** [reply]
Debugging circuits is not my speciality, maybe @vic42 has a suggestion.

But yeah, spikes on 3.3V can be bad, but also depends a lot on exactly where measured. The capacitance of wires means that higher frequency spikes may be hidden, if not smoothed out by decoupling capacitors. Adding such can often be helpful.

**Reply by vic42** [reply, marked as answer]
First of all, the schematic does not state the actual components chosen (especially the inductor and capacitors chosen would matter) . Furthermore, spikes may depend on what is connected on the load side and how. So without looking at the full schematic, the physical layout of the PCB, and the dimensioning calculations for the relevant components, there is not a lot one could recommend. All these tidbits that differentiate a good from a bad design need to be taken into account. (I assumed that you got the LM2596 from a dependable genuine source and not from Aliexpress)

**Reply by martin072** [reply]
Many thanks. Will close this
